### PR TITLE
mgr/vol: remove unused variable

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Class AsyncJob's attribute "self.waiting" is not used anywere in the
codebase. Therefore, remove it.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/volumes/fs/async_job.py [hunk 1]
```diff
@@ -125,8 +125,6 @@ def __init__(self, volume_client, name_pfx, nr_concurrent_jobs):
         # lock, cv for kickstarting jobs
         self.lock = threading.Lock()
         self.cv = threading.Condition(self.lock)
-        # cv for job cancelation
-        self.waiting = False
```
Comment (rishabh-d-dave) [inline src/pybind/mgr/volumes/fs/async_job.py:129]: Before this change -
```
$ pwd
ceph/src/pybind/mgr/volumes/fs
$ grep -rn "waiting" ./
./async_job.py:129:        self.waiting = False
./async_job.py:161:                    # Decrease concurrency: notify threads which are waiting for a job to terminate.
./async_job.py:263:                log.debug("waiting for {0} in-progress jobs for volume '{1}' to "
```

After this change -
```
$ pwd
ceph/src/pybind/mgr/volumes/fs
$ grep -rn "waiting" ./
./async_job.py:159:                    # Decrease concurrency: notify threads which are waiting for a job to terminate.
./async_job.py:261:                log.debug("waiting for {0} in-progress jobs for volume '{1}' to "
```

**Comment by rishabh-d-dave** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68419](https://tracker.ceph.com/issues/68419).

**Review by joscollin — APPROVED** [review]

**Comment by joscollin** [comment]
jenkins test make check

**Review by rishabh-d-dave — COMMENTED** [review]
QA run was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#wip-rishabh-testing-20241007102024

Ready for merge, waiting for CI.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
Failure reason was unrelated. https://jenkins.ceph.com/job/ceph-pull-requests/145649/

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test signed

**Comment by rishabh-d-dave** [comment]
jenkins test licenses

**Pull request merged**